scenario/state.py

```diff
@@ -947,6 +947,8 @@ class _CharmSpec(_DCBase):
     """Charm spec."""
 
     charm_type: Type["CharmType"]
+
+    # TODO: consider unifying the data model since nowadays it's all in one file
```

Comment:
I think it's fine to leave the model as is (and drop this comment): they're in one file now, but they're still logically separate things.

Reply:
If the long-term intention is to leave them separate, then I wonder if `meta` should have `actions` and `config` popped out rather than copied?

As things are now, you can do either of these to access the same data: `meta["actions"][action].description` or `actions[action].description`. It feels like it would be better to remove the chance of ending up with a mixture of code by picking one.

Reply:
yeah was doubting about that one. But you're right, it makes more sense.